app/frontend/components/shared/permit-applications/requirement-form.tsx

```diff
@@ -10,13 +9,14 @@ interface IRequirementFormProps {
 }
 
 export function RequirementForm({ permitApplication, requirements }: IRequirementFormProps) {
-  const [form, setForm] = useState(permitApplication ? permitApplication.requirements : requirements)
-  // const [form, setForm] = useState(stubbedForm)
+  const form = permitApplication ? permitApplication.requirements : requirements
+  const [submission, setSubmission] = useState(permitApplication?.submissionData || {})
```

Comment:
why is a separate state needed? mobx should be reactive if you change the function to observers

Reply:
One is for loading the 'previous' submission to restore saved state.  The setSubmission is being used when the file gets saved in the backend and the payload needs to get reformatted.  Will leave a nte maybe to see if this is working as expected.

Reply:
@sunto but that should happen anyway if you update the submissionData on mobx, unless you are planning to do optimistic updates, this is not necessary. Even then you can just do this from within the action. Lemme know if you want to chat about this